src/components/navigation/Navigation.jsx

```diff
@@ -12,32 +12,72 @@ const Navigation = () => {
     cartState: { cartList },
   } = useCartContext();
 
+  // getting encodedToken from localStorage
+  const encodedToken = localStorage.getItem("token");
+
+  // from react-router-dom
+  const path = useLocation();
+
   return (
     <header className="header">
       <div className="navbar-container">
         <NavLink className="navbar" to="/">
           <h1 className="navbar-m-lr-1">LUCENT STORE</h1>
         </NavLink>
-        <NavLink className="navbar-m-left-auto navbar" to="/login">
-          <h5 className="navbar-m-lr-1">LOGIN</h5>
-        </NavLink>
-        <div className="search-bar-border navbar-m-lr-1">
-          <input className="input search-bar-input" type="text" />
-          <i className="fas fa-search"></i>
-        </div>
-        <NavLink className="navbar navbar-m-lr-1" to="/products">
-          <i className="fa-solid fa-store"></i>
-        </NavLink>
-        <NavLink to="/wishlist">
-          <IconWithBadge count={wishlist.length}>
-            <i className="fas fa-heart icon-badge"></i>
-          </IconWithBadge>
-        </NavLink>
-        <NavLink to="/cart">
-          <IconWithBadge className="m-r-2rem" count={cartList.length}>
-            <i className="fas fa-shopping-cart icon-badge"></i>
-          </IconWithBadge>
-        </NavLink>
+        {!encodedToken &&
+          path.pathname !== "/logout" &&
+          path.pathname !== "/signin" &&
```

Comment:
suggestion: you can use context to store user and check if user exist, then you won't have to check for routes like this.